client/package.json

```diff
@@ -4,19 +4,23 @@
   "license": "MIT",
   "angular-cli": {},
   "scripts": {
-    "update-shared": "(cd ../shared && node build.js)",
     "start": "ng serve",
+    "build": "ng build --prod",
+    "build:staging": "ng build --env=staging --prod --sourcemaps false --progress false",
+    "build-and-serve": "npm run build && serve -s dist/",
+    "build-and-serve:staging": "npm run build:staging && serve -s dist/",
+    "test": "ng test",
+    "test:single-run": "ng test --single-run",
+    "e2e": "cypress run",
     "lint": "ng lint",
+    "lint-and-check": "npm run lint && npm run format:check",
     "prettier": "prettier --config ../.prettierrc --ignore-path ../.prettierignore",
     "format:base": "npm run prettier -- '{src,e2e}/**/*{.ts,.js,.json,.css,.scss}'",
     "format:check": "npm run format:base -- --list-different",
     "format:fix": "npm run format:base -- --write",
-    "ci": "npm run lint && npm run format:check && npm run test && npm run test:e2e:production",
-    "test": "ng test --single-run && ng e2e --env=e2e",
-    "test:e2e:production": "ng e2e --target=production --env=e2e",
-    "pree2e": "webdriver-manager update",
-    "e2e": "ng e2e --serve false",
-    "node_modules": "rm -rf ./node_modules && yarn install"
+    "ci": "npm run lint-and-check && npm run test:single-run && CYPRESS_baseUrl=http://localhost:5000 start-server-and-test build-and-serve:staging http://localhost:5000 e2e",
```

Comment:
Notice that I'm overriding cypress `baseUrl` here with an environment variable. This is to ensure that in the `ci` script, cypress looks for `localhost:5000`.

There might be a way to configure env variables in travis, but then the `ci` script wouldn't work locally anymore, unless the env variable is set manually.

No strong opinion here.